refactor: fix mypy issues in checkers/c*.py

Fix mypy errors for the following:

```
cve_bin_tool/checkers/cvs.py:17: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/cvs.py:18: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/cronie.py:15: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/connman.py:16: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/connman.py:17: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/collectd.py:15: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/collectd.py:16: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/clamav.py:17: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/clamav.py:18: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/chrony.py:16: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/chrony.py:17: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/chess.py:15: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/chess.py:16: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/c_ares.py:16: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/c_ares.py:17: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
```

Helpful resources:

- [Python typing documentation](https://docs.python.org/3/library/typing.html)
- [mypy cheat sheet](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)
- If you can't find the answer in the docs, googling the error messages above will often help!
- [cve-bin-tool's contributor docs](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md)
- If you've contributed to open source but not this project, you might just want our [checklist for a great pull request](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md#checklist-for-a-great-pull-request)

Tiny tips for new contributors:

- cve-bin-tool uses https://www.conventionalcommits.org/ style for commit messages, and we have a test that checks the title of your PR. A good potential title for this one is in the title of this issue.
- You can make an issue auto close by including a comment "fixes #ISSUENUMBER" in your PR comments where ISSUENUMBER is the actual number of the issue.

--- a/cve_bin_tool/checkers/c_ares.py
+++ b/cve_bin_tool/checkers/c_ares.py
@@ -9,12 +9,14 @@ https://www.cvedetails.com/product/11384/Daniel-Stenberg-C-ares.html?vendor_id=6
 https://www.cvedetails.com/product/34754/C-ares-Project-C-ares.html?vendor_id=15926
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class CAresChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = [
         r"c-ares-([0-9]+\.[0-9]+\.[0-9]+)",
         r"c-ares library initialization[A-Za-z \.\r\n]+\r?\n([0-9]+\.[0-9]+\.[0-9]+)",

--- a/cve_bin_tool/checkers/chess.py
+++ b/cve_bin_tool/checkers/chess.py
@@ -8,11 +8,13 @@ CVE checker for chess
 https://www.cvedetails.com/product/1867/GNU-Chess.html?vendor_id=72
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class ChessChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = [r"Chess ([0-9]+\.[0-9]+\.[0-9]+)"]
     VENDOR_PRODUCT = [("gnu", "chess")]

--- a/cve_bin_tool/checkers/chrony.py
+++ b/cve_bin_tool/checkers/chrony.py
@@ -9,12 +9,14 @@ https://www.cvedetails.com/product/18821/Tuxfamily-Chrony.html?vendor_id=10533
 https://www.cvedetails.com/product/64144/Chrony-Project-Chrony.html?vendor_id=21263
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class ChronyChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = [
         r"([0-9]+\.[0-9]+)\r?\nchrony",
         r"chrony\r?\n([0-9]+\.[0-9]+)",

--- a/cve_bin_tool/checkers/clamav.py
+++ b/cve_bin_tool/checkers/clamav.py
@@ -10,12 +10,14 @@ https://www.cvedetails.com/product/15657/Clamav-Clamav.html?vendor_id=8871
 https://www.cvedetails.com/product/17215/Clamavs-Clamav.html?vendor_id=9637
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class ClamavChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = [
         r"clamav-([0-9]+\.[0-9]+\.[0-9]+)",
         r"([0-9]+\.[0-9]+\.[0-9]+)\r?\nUnable to allocate memory for db directory...\r?\n%s/daily.cvd\r?\n%s/daily.cld\r?\nClamAV %s",

--- a/cve_bin_tool/checkers/collectd.py
+++ b/cve_bin_tool/checkers/collectd.py
@@ -8,11 +8,13 @@ CVE checker for collectd
 https://www.cvedetails.com/product/20310/Collectd-Collectd.html?vendor_id=11242
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class CollectdChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = ["collectd ([0-9]+\\.[0-9]+\\.[0-9]+)"]
     VENDOR_PRODUCT = [("collectd", "collectd")]

--- a/cve_bin_tool/checkers/connman.py
+++ b/cve_bin_tool/checkers/connman.py
@@ -9,11 +9,13 @@ https://www.cvedetails.com/product/22328/Connman-Connman.html?vendor_id=11940
 https://www.cvedetails.com/product/63425/Intel-Connman.html?vendor_id=238
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class ConnmanChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = [r"([0-9]+\.[0-9]+)\r?\nGWeb/%s"]
     VENDOR_PRODUCT = [("connman", "connman"), ("intel", "connman")]

--- a/cve_bin_tool/checkers/cronie.py
+++ b/cve_bin_tool/checkers/cronie.py
@@ -8,11 +8,13 @@ CVE checker for cronie
 https://www.cvedetails.com/product/18871/?q=Cronie
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class CronieChecker(Checker):
-    CONTAINS_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
     FILENAME_PATTERNS = [r"cronnext", r"crontab", r"crond"]
     VERSION_PATTERNS = [r"cronie ([0-9]+\.[0-9]+\.[0-9]+)"]
     VENDOR_PRODUCT = [("fedorahosted", "cronie")]

--- a/cve_bin_tool/checkers/cvs.py
+++ b/cve_bin_tool/checkers/cvs.py
@@ -10,11 +10,13 @@ https://www.cvedetails.com/product/20192/Nongnu-CVS.html?vendor_id=6788
 https://www.cvedetails.com/product/40040/GNU-CVS.html?vendor_id=72
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class CvsChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = [r"CVS\) ([0-9]+\.[0-9]+\.[0-9]+)"]
     VENDOR_PRODUCT = [("cvs", "cvs"), ("gnu", "cvs"), ("nongnu", "cvs")]

